_transcripts/032-rust-1.48-1.49.md

```diff
@@ -0,0 +1,731 @@
+here in this podcast, because in the release notes, usually it's overshadowed
+by, you know, language changes and library changes. But we should definitely
+shout out to it a bit more. It's an unsung hero of the Rust world.
+
+__Jon__: Yeah.
+
+__Ben__: Can't live without the documentation. And there's one more change too,
+to `rustdoc` in this release.
+
+__Jon__: Yeah, I was pretty happy to see `rustdoc` get some love here. Do you
+want to talk about this new `#[doc(alias)]` feature?
+
+__Ben__: Sure. And so pretty much one of the cool features of `rustdoc` is it
+has a built in search bar at the very top there, which, if you're not using it,
+you should be. And sometimes it can be hard to find exactly what you're looking
+for if you don't quite remember what you were— like, what the name of the thing
+is. And so now in this new release, you can, on top of any item that you want to
+document, you can give it an alias, and you can say, hey, someone might be
+misremembering the name of this, or they might mis-type it, do a typo. And so
+you can say, hey, like, if they search for this, show them this instead. So it's
+just a nice little quality of life thing, for making the docs a bit easier to
+navigate.
+
+At some point, we should also go through and, like, talk about other cool ways
+to use the advanced features of `rustdoc` searching, like, you can search for
+return types or function types. If you know that some function returns a
+`String`, you can just be like, hey, I want to find all things around here that
+return a `String`. Oh, here's what I wanted in the first place. So even if you
+don't know the name of the function or if one even exists, you can kind of
+explore the standard library, which is pretty big now, kind of getting large
+with all these convenience types, and find the thing that you're looking for in
+there.
+
+__Jon__: Yeah, it's funny, because in some sense, this is like, how Haskell
+documentation to some extent works, is that the function signature is a big part
+of the documentation, and that is sort of true in Rust as well. Like in many
+cases, you can give me a function signature without the function name, and I
+might be able to tell you what that function does just based on the types
+involved. And `rustdoc` sort of lets you search in that way, to say, I want a
+function that, like returns this type and takes arguments of this type, and that
+might get you pretty far.
+
+__Ben__: In the release notes here, it talks about how an obvious use of this is
+for, if you are doing FFI. If you're writing wrappers for C crates, where often
+C types and Rust types might have different name, based on, you know, the naming
+conventions of the platform or the language. And so in this case you could, say,
+search for `memcpy`, and then get the Rust equivalent, which I believe is "copy
+overlapping" or one of those things. _(editor's note: `copy_nonoverlapping`.)_
```

Comment:
I'll point out that I'm editorializing, though I think the correction should be non-controversial.